app/models/providers/infra/host.rb

```diff
@@ -21,6 +21,10 @@ class Infra::Host < ApplicationRecord
     has_many   :vms, :inverse_of => :host
     has_many   :templates, :inverse_of => :host
 
+    def disconnects
+      vms_and_templates
```

Comment:
Hm this won't will only work for save_vms_inventory, if it is in save_hosts_inventory and the target was the host then the disconnect should be `self.clone`